chainer/utils/__init__.py

```diff
@@ -1,4 +1,4 @@
-import collections
+import collections as _collections
```

Comment:
I just noticed; does it work correctly on py2? It would be better to use `from __future__ import absolute_import` to avoid conflicts with `./collections.py`.